diff --git a/Makefile.in b/Makefile.in
index e692405..7451745 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3,7 +3,7 @@ SRC = md2c.c md4c.c md5c.c md2hl.c md4hl.c md5hl.c\
 OBJ = $(SRC:.c=.o)
 
 CC      = @CC@
-CFLAGS  = -O3 -fPIC -I.
+CFLAGS += -fPIC -I.
 RANLIB  = @RANLIB@
 AR      = ar
 LN_S    = @LN_S@
@@ -31,18 +31,18 @@ docs:
 test:	mddriver
 	./mddriver
 
-install:	libmd.so.1.0 libmd.so.1 libmd.so
+install:	libmd.so.1.0
 	mkdir -p $(BUILDROOT)/usr/include
 	mkdir -p $(BUILDROOT)/usr/lib
 	mkdir -p $(BUILDROOT)/usr/man/man3
-	$(INSTALL) -g root -o root -m 644 md2.h md4.h md5.h sha.h rmd160.h \
+	$(INSTALL) -m 644 md2.h md4.h md5.h sha.h rmd160.h \
 		   $(BUILDROOT)/usr/include
-	$(INSTALL) -g root -o root -m 644 md2.3 md4.3 md5.3 \
+	$(INSTALL) -m 644 md2.3 md4.3 md5.3 \
 		   $(BUILDROOT)/usr/man/man3
-	$(INSTALL) -g root -o root -m 755 libmd.so.1.0 $(BUILDROOT)/usr/lib
-	$(INSTALL) -g root -o root -m 755 libmd.so.1 $(BUILDROOT)/usr/lib
-	$(INSTALL) -g root -o root -m 755 libmd.so $(BUILDROOT)/usr/lib
-	$(INSTALL) -g root -o root -m 755 libmd.a $(BUILDROOT)/usr/lib
+	$(INSTALL) -m 755 libmd.a $(BUILDROOT)/usr/lib
+	$(INSTALL) -m 755 libmd.so.1.0 $(BUILDROOT)/usr/lib
+	$(LN_S) libmd.so.1.0 $(BUILDROOT)/usr/lib/libmd.so.1
+	$(LN_S) libmd.so.1.0 $(BUILDROOT)/usr/lib/libmd.so
 
 binary:	install
 	@echo Creating binary .tar.bz2 in directory $(OUTDIR)/
@@ -54,12 +54,6 @@ tarball: mrproper
 	cd .. && tar cf $(OUTDIR)/libmd-$(VERS).tar libmd-$(VERS)/
 	bzip2 -9f $(OUTDIR)/libmd-$(VERS).tar
 
-libmd.so.1:
-	$(LN_S) libmd.so.1.0 libmd.so.1
-
-libmd.so:
-	$(LN_S) libmd.so.1.0 libmd.so
-
 md2hl.c: mdXhl.c
 	sed -e 's/mdX/md2/g' -e 's/MDX/MD2/g' $? > $@
 
@@ -88,7 +82,7 @@ md5.3:	mdX.3
 	cat md5.copyright >> $@
 
 libmd.so.1.0:	$(OBJ)
-	$(CC) -shared -o libmd.so.1.0 $(OBJ)
+	$(CC) -Wl,-soname,libmd.so.1 -shared -o libmd.so.1.0 $(OBJ)
 
 libmd.a:	$(OBJ)
 	rm -f libmd.a
