diff --git a/config/kernel.m4 b/config/kernel.m4
index 7c1566d..7b0f47b 100644
--- a/config/kernel.m4
+++ b/config/kernel.m4
@@ -375,7 +375,7 @@ AC_DEFUN([ZFS_LINUX_COMPILE_IFELSE], [
 	rm -Rf build && mkdir -p build
 	echo "obj-m := conftest.o" >build/Makefile
 	AS_IF(
-		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-Werror-implicit-function-declaration $EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build) >/dev/null && AC_TRY_COMMAND([$3])],
+		[AC_TRY_COMMAND(cp conftest.c build && make [$2] -C $LINUX_OBJ EXTRA_CFLAGS="-fplugin-arg-constify_plugin-no-constify -Werror-implicit-function-declaration $EXTRA_KCFLAGS" $ARCH_UM M=$PWD/build) >/dev/null && AC_TRY_COMMAND([$3])],
 		[$4],
 		[_AC_MSG_LOG_CONFTEST m4_ifvaln([$5],[$5])]
 	)
