diff --git a/sys/boot/i386/boot2/Makefile b/sys/boot/i386/boot2/Makefile
index 701d30f..0f51738 100644
--- a/sys/boot/i386/boot2/Makefile
+++ b/sys/boot/i386/boot2/Makefile
@@ -41,7 +41,7 @@ CFLAGS=	-Os \
 	-Winline --param max-inline-insns-single=100
 
 .if ${CC:T:Mclang} == "clang"
-CFLAGS+=	-mllvm -stack-alignment=8 -mllvm -inline-threshold=3 \
+CFLAGS+=	-mstack-alignment=8 -mllvm -inline-threshold=3 \
 		-mllvm -enable-load-pre=false
 # XXX: clang integrated-as doesn't grok .codeNN directives yet
 CFLAGS+=	${.IMPSRC:T:Mboot1.S:C/^.+$/-no-integrated-as/}
